### PR TITLE
Protect: Add unignore threat capabilities

### PR DESCRIPTION
**Pull request by dkmyta**
## Description
Allows users to unignore threats found in the ignored section of their scan history.

Reliant on:
- Automattic/vaultpress#106996
- D153672

## Proposed changes:

* Adds an endpoint responsible for calling the wpcom endpoint to trigger the unignore change
* Adds a conditional unignored button to ignored threats in scan history UI

### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1250

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch and start up Jurassic Tube
* Install Protect add fixable and non-fixable threats and run a scan to detect them
* Ignore several threats and proceed to History and filter by `Ignored`
* Select `Unignore threat` and verify that the threat is added back to the current threats list
* Check `vp-global` to ensure the `vp_scan_results` entry was updated as expected
* Ensure no regressions are introduced

### CODE REVIEW

**Review by nateweller — CHANGES_REQUESTED** [review]
Needs a rebase

File: projects/plugins/protect/src/js/components/unignore-threat-modal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+				{ createInterpolateElement(
+					__(
+						'By choosing to unignore this threat, you acknowledge that you have reviewed the detected code. You are accepting the risks of maintaining a potentially malicious or vulnerable file on your site. If you are unsure, please request an estimate with <codeableLink>Codeable</codeableLink>.',
+						'jetpack-protect'
+					),
+					{
+						codeableLink: <Button variant="link" isExternalLink={ true } href={ codeableURL } />,
+					}
+				) }
+			</Text>
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/unignore-threat-modal/index.jsx]:
```suggestion
```

What do you think about removing this blurb for the unignore step? Seems less relevant.

**Review by nateweller — APPROVED** [review]
Works well and code looks good! :shipit:

File: projects/plugins/protect/src/js/components/unignore-threat-modal/index.jsx (hunk 1 above)
Comment (dkmyta) [reply projects/plugins/protect/src/js/components/unignore-threat-modal/index.jsx]: Agreed!

**Pull request merged**